src/lazy/any_encoding.rs

```diff
@@ -300,6 +300,24 @@ impl IonEncoding {
         use IonEncoding::*;
         matches!(*self, Binary_1_0 | Binary_1_1)
     }
+
+    pub fn name(&self) -> &str {
+        use IonEncoding::*;
+        match self {
+            Text_1_0 => TextEncoding_1_0::name(),
+            Binary_1_0 => BinaryEncoding_1_0::name(),
+            Text_1_1 => TextEncoding_1_1::name(),
+            Binary_1_1 => BinaryEncoding_1_1::name(),
+        }
+    }
+
+    pub fn version(&self) -> (u8, u8) {
+        use IonEncoding::*;
+        match self {
+            Text_1_0 | Binary_1_0 => (1, 0),
+            Text_1_1 | Binary_1_1 => (1, 1),
+        }
+    }
```

Comment:
🗺️ These methods make it easier for programs like `ion inspect` to check what encoding they're working with and construct error messages.